doc/usecases/core/UC_keyset_remove.md

```diff
@@ -12,13 +12,17 @@
 - **Precondition:**
   - [`KeySet` has been created](UC_keyset_create.md)
 - **Main success scenario:**
-  - Caller requests to remove `Key` by name from `KeySet`
-  - Core searches for `Key` with same name in `KeySet`
-  - If a matching `Key` is found, Core removes it from `KeySet` returns a `Key *` to it.
-    The name of the `Key` will be read-only, otherwise it is modifiable.
-  - Otherwise, Core returns `NULL`
-- **Alternative scenario:** -
-- **Error scenario:** -
+  - Caller requests to remove `Key` by index from `KeySet`
+  - Index is in-range for `KeySet`
+  - Core removes `Key` at index and resizes `KeySet`.
+- **Alternative scenario:**
+  - Caller requests to remove a range of `Key`s by index range from `KeySet`
```

Comment:
This probably can be main scenario, as the range simply would be 1. If there is a convenience function for that, is imho not really relevant for the use case. But again, there should be a motivation why a user wants this. (E.g. cut out a hierarchy).

Reply:
That's true, I'll swap main and alternative scenario.

But do we actually need a detailed motivation here? Removing range of elements in a sorted data structure seems pretty basic too me. For example, the entire `KeySet` might be an array (e.g. extracted from elsewhere) and you want to remove a section of the array.

Reply:
If we were happy without it up to now, there should be definitely some motivation now. The best motivation would be something from LCDproc or a plugin (something practical in the real world); together with a non-functional requirement.

Reply:
The thing is "happy until now" is hard to define. You can always call the function "remove Key at index X" repeatedly in a loop to remove N keys. So maybe somebody already encountered a practical use case. But OTTOMH I don't know a motiviation for general ranges.

But for hierarchies, we just need to look at were `ksCut` is used. All of those become a use case for "remove index range", with the index range returned from "find hierarchy". That the range doesn't have to be hierarchy is just neat (and free) bonus IMO. I'll add a note about that.

The non-functional requirement (which I'll add) is already covered in https://github.com/ElektraInitiative/libelektra/pull/4797#discussion_r1060901517. I just noticed that over there I actually explain "remove range" not "access index range". But the logic is the same, the numbers are just a bit different. Asymptotically the range functions are better than calling the "normal" functions in a loop.